src/cargo/ops/cargo_doc.rs

```diff
@@ -24,7 +24,17 @@ pub fn doc(ws: &Workspace<'_>, options: &DocOptions) -> CargoResult<()> {
             .root_crate_names
             .get(0)
             .ok_or_else(|| anyhow::anyhow!("no crates with documentation"))?;
-        let kind = options.compile_opts.build_config.single_requested_kind()?;
+        let pkg = ws
+            .members()
+            .find(|p| p.name().to_string() == *name.clone().replace("_", "-"))
+            .unwrap();
+        let kind = if let Some(forced_kind) = pkg.manifest().forced_kind() {
```

Comment:
Should we respect `--target` here?

Reply:
Also, if we only try one package.
1. cargo clean
2. `cargo doc --open --target aarch64-apple-darwin  -p bar `
3. It generated:
```console
➜  target git:(main) ✗ tree -L 2
.
├── CACHEDIR.TAG
├── aarch64-apple-darwin
│   ├── CACHEDIR.TAG
│   └── debug
├── debug
│   ├── build
│   ├── deps
│   ├── examples
│   └── incremental
└── riscv64gc-unknown-none-elf
    ├── CACHEDIR.TAG
    ├── debug
    └── doc
```
4. It didn't open any docs.